modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -560,6 +578,19 @@ public CompletableFuture<Boolean> notify(SchemaEventParameters parameters, @Null
         addMessageHandler(clusterService.messagingService());
     }
 
+    private CompletableFuture<Void> performRebalanceOnRecovery(long revision) {
+        var startRange = new ByteArray(PENDING_ASSIGNMENTS_PREFIX);
+        var endRange = new ByteArray(incrementLastChar(PENDING_ASSIGNMENTS_PREFIX));
```

Comment:
Maybe I didn't understand something, but we can receive `IllegalArgumentException` here, if last char is the `Character.MAX_VALUE`. But I think instead we want to have a `new ByteArray(PENDING_ASSIGNMENTS_PREFIX).getBytes(StandardCharsets.UTF_8) ++ [Character.MIN_VALUE])`to achieve the our goal (where ++ is a pseudocode for arrays concatenation).

Reply:
> Maybe I didn't understand something, but we can receive IllegalArgumentException here, if last char is the Character.MAX_VALUE

This is true, but should we bother in the real world? Who is going to use Character.MAX_VALUE in their strings?

> But I think instead we want to have a new ByteArray(PENDING_ASSIGNMENTS_PREFIX).getBytes(StandardCharsets.UTF_8) ++ [Character.MIN_VALUE])

This is not true, this will result in an empty range